engine/gameobject/src/dmsdk/gameobject/gameobject.h

```diff
@@ -250,10 +252,12 @@ namespace dmGameObject
      * @struct
      * @name PropertyOptions
      * @member m_Index [type:int32_t] The index of the property to set, only applicable if property is array.
+     * @member m_Key [type:char*] The key of the property to set, only applicable if property is hashtable.
      */
     struct PropertyOptions
     {
         int32_t m_Index;
+        dmhash_t m_Key;
     };
```

Comment:
Given that a property _might_ support both ways of indexing, we need a way to distinguish between the two.

E.g.:
```
    struct PropertyOptions
    {
        union {
            dmhash_t m_Key;
            int32_t m_Index;
        }; // It now takes 8 bytes and not 16

        uint8_t m_HasKey:1: // it takes 9 bytes with this
    };
```

Reply:
Done! thx!